lib/functions/RegularFunctions.ts

```diff
@@ -641,17 +783,15 @@ const rand = {
 // https://www.w3.org/TR/sparql11-query/#func-date-time
 // ----------------------------------------------------------------------------
 
-function parseDate(dateLit: E.DateTimeLiteral): P.ISplittedDate {
-  return P.parseXSDDateTime(dateLit.str());
-}
-
 /**
  * https://www.w3.org/TR/sparql11-query/#func-now
  */
 const now = {
   arity: 0,
   overloads: declare(C.RegularOperator.NOW).set([], (sharedContext: ICompleteSharedContext) => () =>
-    new E.DateTimeLiteral(sharedContext.now, sharedContext.now.toISOString())).collect(),
+    new E.DateTimeLiteral(toDateTimeRepresentation(
+      { date: sharedContext.now, timeZone: sharedContext.defaultTimeZone },
```

Comment:
Does it really change the behaviour in a breaking way?
I guess it just enhances it with the default timezone, right?

Reply:
Right now, when a date is given to the `now` context item, the timezone is always there since a date created in JS will have your timezone. When we use this code, the timeZone will be overwritten by the implicit time zone UTC by default). This behavior could potentially break code for our users. 

The issue could be solved if we don't default to UTC when a dateTime is not provided but just use the undefined in our context. This function could then check if it was provided and based on that use the users time zone.

The other functions could still default to UTC (which is in those function definitely what we want for stability reasons). I don't know what would be best here since having special treatments for one function also isn't pretty?

Reply:
Perhaps defaulting the timezone to UTC is the problem.
Maybe we just always want to default this to `now`, which is either user-provided, or system-default.

Reply:
It could be seen as the problem, yes.
The issue is you want this to be kind of stable. What this means is that you don't want your evaluation of:
`"2023-03-01T15:00" = "2023-03-01T15:00Z"` to change just by entering a new time zone (and even weirder, day time saving)

Reply:
> What this means is that you don't want your evaluation of:
"2023-03-01T15:00" = "2023-03-01T15:00Z" to change just by entering a new time zone

But that is exactly what we _do_ want to happen when changing the timezone, right?

Reply:
That's my interpretation of the spec at least, but if you have a different interpretation, do let me know, because I could be wrong of course :-)